functests/1_performance/cpu_management.go

```diff
@@ -483,9 +511,78 @@ var _ = Describe("[rfe_id:27363][performance] CPU Management", func() {
 			}
 		})
 	})
+
+	When("strict NUMA aligment is requested", func() {
+		var testpod *corev1.Pod
+
+		BeforeEach(func() {
+			if profile.Spec.NUMA == nil || profile.Spec.NUMA.TopologyPolicy == nil {
+				Skip("Topology Manager Policy is not configured")
+			}
+			tmPolicy := *profile.Spec.NUMA.TopologyPolicy
+			if tmPolicy != "single-numa-node" {
+				Skip("Topology Manager Policy is not Single NUMA Node")
+			}
+		})
+
+		AfterEach(func() {
+			if testpod == nil {
+				return
+			}
+			deleteTestPod(testpod)
+		})
+
+		It("should reject pods which request integral CPUs not aligned with machine SMT level", func() {
+			cpuCount := 1 // not must be intentionally < than the smtLevel
+			testpod = promotePodToGuaranteed(getStressPod(workerRTNode.Name, cpuCount))
+			testpod.Namespace = testutils.NamespaceTesting
+
+			smtLevel := getSMTLevel(workerRTNode)
+			if smtLevel < 2 {
+				Skip(fmt.Sprintf("designated worker node %q has SMT level %d - minimum required 2", workerRTNode.Name, smtLevel))
+			}
+
+			err := testclient.Client.Create(context.TODO(), testpod)
+			Expect(err).ToNot(HaveOccurred())
+
+			err = pods.WaitForPredicate(testpod, 10*time.Minute, func(pod *corev1.Pod) (bool, error) {
+				if pod.Status.Phase != corev1.PodPending {
+					return true, nil
+				}
+				return false, nil
+			})
+			Expect(err).ToNot(HaveOccurred())
+
+			updatedPod := &corev1.Pod{}
+			err = testclient.Client.Get(context.TODO(), client.ObjectKeyFromObject(testpod), updatedPod)
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(updatedPod.Status.Phase).To(Equal(corev1.PodFailed), "pod %s not failed: %v", updatedPod.Name, updatedPod.Status)
+			Expect(isSMTAlignmentError(updatedPod)).To(BeTrue(), "pod %s failed for wrong reason: %q", updatedPod.Name, updatedPod.Status.Reason)
+		})
+	})
+
 })
 
-func getStressPod(nodeName string) *corev1.Pod {
+func isSMTAlignmentError(pod *corev1.Pod) bool {
+	re := regexp.MustCompile(`SMT.*Alignment.*Error`)
+	return re.MatchString(pod.Status.Reason)
+}
+
+func getSMTLevel(node *corev1.Node) int {
+	cpuID := 0 // this is just the most likely cpu to be present in a random system. No special meaning besides this.
```

Comment:
hm can we just call `lscpu` and check `Thread(s) per core:  2`, like in this case we will have less assumptions regarding CPU ids.

Reply:
I've been trying hard to require `lscpu` for this task to minimize the deps - I'm already uncomfortable using `tr` when the info is inside files `sysfs`. I guess I'll review this logic and try to simplify.

Reply:
I see, I just thought because we are using some pre-defined container image we can have some assumptions

Reply:
correct, to some extent. But still let's try to minimize the assumptions!